packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkConfigurationExt.kt

```diff
@@ -75,31 +76,108 @@ internal fun ReadableMap.asProxyConfig(): Pair<Proxy, ProxyAuthenticator?>? {
+    const val nativeLongTaskThresholdMs = 200.0
+    const val nativeViewTracking = false
+    const val nativeInteractionTracking = false
+    const val trackingConsent = "GRANTED"
+    const val telemetrySampleRate = 20.0
+    const val vitalsUpdateFrequency = "AVERAGE"
+    const val trackFrustrations = true
+    const val uploadFrequency = "AVERAGE"
+    const val batchSize = "MEDIUM"
+    const val trackBackgroundEvents = false
+}
+
+@Suppress("ComplexMethod")
+internal fun JSONDdSdkConfiguration.asDdSdkConfiguration(): DdSdkConfiguration {
+    return DdSdkConfiguration(
+        this.clientToken,
+        this.env,
+        this.applicationId,
+        this.nativeCrashReportEnabled ?: DefaultConfiguration.nativeCrashReportEnabled,
+        this.nativeLongTaskThresholdMs ?: DefaultConfiguration.nativeLongTaskThresholdMs,
+        this.longTaskThresholdMs ?: DefaultConfiguration.longTaskThresholdMs,
+        this.sessionSamplingRate ?: DefaultConfiguration.sessionSamplingRate,
+        this.site ?: DefaultConfiguration.site,
+        this.trackingConsent ?: DefaultConfiguration.trackingConsent,
+        this.telemetrySampleRate ?: DefaultConfiguration.telemetrySampleRate,
+        this.vitalsUpdateFrequency ?: DefaultConfiguration.vitalsUpdateFrequency,
+        this.trackFrustrations ?: DefaultConfiguration.trackFrustrations,
+        this.uploadFrequency ?: DefaultConfiguration.uploadFrequency,
+        this.batchSize ?: DefaultConfiguration.batchSize,
+        this.trackBackgroundEvents ?: DefaultConfiguration.trackBackgroundEvents,
+        this.customEndpoints,
+        mapOf(
+            "_dd.source" to "react-native",
+            "_dd.sdk_version" to SDK_VERSION
+        ),
+        null,
+        this.nativeViewTracking ?: DefaultConfiguration.nativeViewTracking,
+        this.nativeInteractionTracking ?: DefaultConfiguration.nativeInteractionTracking,
+        this.verbosity,
+        this.proxy?.asProxyConfig(),
+        this.serviceName,
+        this.firstPartyHosts?.asFirstPartyHosts()
+    )
+}
+
+internal fun JSONProxyConfiguration.asProxyConfig(): Pair<Proxy, ProxyAuthenticator?>? {
+    val proxyType = when (type.lowercase(Locale.US)) {
                 "http", "https" -> Proxy.Type.HTTP
                 "socks" -> Proxy.Type.SOCKS
                 else -> {
                     Log.w(
                         DdSdk::class.java.canonicalName,
-                        "Unknown proxy type given: $it, skipping proxy configuration."
+                        "Unknown proxy type given: $type, skipping proxy configuration."
                     )
                     null
                 }
             }
-        }
 
     val proxy =
-        if (address != null && port != null && type != null) {
-            Proxy(type, InetSocketAddress(address, port))
+        if (proxyType != null) {
+            Proxy(proxyType, InetSocketAddress(address, port))
         } else {
             return null
```

Comment:
nit: It's easy to look at this and think `proxy` gets assigned `null` when in reality the function is returning `null`. An early out might read better, and Kotlin should be smart enough with nulls so you don't have to force dereference:

```kotlin
if (proxyType == null) {
    return null
}

val proxy = Proxy(proxyType, InetSocketAddress(address, port))
```